### PR TITLE
Add WebTransport support.

### PR DESCRIPTION
**Pull request by jianjunz**
This PR adds WebTransport - QuicTransport server for conference mode. It depends on https://github.com/open-webrtc-toolkit/owt-server/pull/113.

### CODE REVIEW

File: src/samples/conference/README.md [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
 2. Edit `samplertcservice.js`. Find `icsREST.API.init`, replace service ID, service key and REST server URL with the correct values.
 3. Copy your SSL server certificate to cert/certificate.pfx.
 4. Run `initcert.js` to generate a keystore file which contains encrypted SSL server's passphase.
 5. Run `node samplertcservice.js` to start conference sample. By default, it listens on port 3001(insecure) and 3004(secure).
-6. Open a browser and navigate to http://\<hostname\>:3001/ or https://\<hostname\>:3004/.
+6. Open a browser and navigate to http://hostname:3001/ or https://hostname:3004/.
+
+## QUIC
+
+`public\scripts\index.js` was modified to create QUIC connections between client and server. After navigating to https://hostname:3004/?publish=false, the page automatically creates a QuicStream for publishing. It also listens to `stream-added` message and creates new QuicStreams for subscribing. It doesn't subscribe streams published before joining. `window.bidirectionalStream` is the last QuicStream it created. You may write or read data in console by accessing `window.bidirectionalStream`. Please try to modify index.js if you want to try other QUIC features.
```
Comment (taste1981) [inline src/samples/conference/README.md:15]: should be quic.js?

File: src/sdk/conference/client.js [hunk 2]
```diff
@@ -335,6 +346,12 @@ export const ConferenceClient = function(config, signalingImpl) {
    * @param {string} tokenString Token is issued by conference server(nuve).
    */
   this.join = function(tokenString) {
+    if (QuicConnection) {
+      // TODO: Get URL from token.
+      quicTransportChannel = new QuicConnection(
+          'quic-transport://jianjunz-nuc-ubuntu.sh.intel.com:7700/echo',
```
Comment (taste1981) [inline src/sdk/conference/client.js:352]: please remove this.

File: src/sdk/conference/client.js [hunk 3]
```diff
@@ -422,7 +441,16 @@ export const ConferenceClient = function(config, signalingImpl) {
     if (!(stream instanceof StreamModule.RemoteStream)) {
       return Promise.reject(new ConferenceError('Invalid stream.'));
     }
-    const channel = createPeerConnectionChannel();
+    if (stream.source.data) {
```
Comment (taste1981) [inline src/sdk/conference/client.js:444]: I think "data" is not one kind of source. We should update streaminfo strcuture to identify av/data with "kind" or "media_type" instead.

File: src/sdk/conference/client.js [hunk 4]
```diff
@@ -455,4 +483,21 @@ export const ConferenceClient = function(config, signalingImpl) {
       signalingState = SignalingState.READY;
     });
   };
+
+  /**
+   * @function createSendStream
+   * @memberOf Owt.Conference.ConferenceClient
+   * @instance
+   * @desc Create a outgoing stream. Only available when WebTransport is supported by user's browser.
```
Comment (taste1981) [inline src/sdk/conference/client.js:491]: an outgoing.

File: src/sdk/conference/quicconnection.js [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    const writer = quicStream.writable.getWriter();
+    await writer.ready;
+    writer.write(this._uuidToUint8Array(publicationId));
+    writer.releaseLock();
+    this._quicStreams.set(publicationId, quicStream);
+    return quicStream;
+  }
+
+  async publish(stream) {
+    // TODO: Avoid a stream to be published twice. The first 16 bit data send to
```
Comment (taste1981) [inline src/sdk/conference/quicconnection.js:137]: I think this is a workaround. Can we remove that requirement.

File: src/samples/conference/README.md (hunk 1 above)
Comment (jianjunz) [reply src/samples/conference/README.md:15]: Yes, it's moved to a new file.

File: src/sdk/conference/client.js (hunk 3 above)
Comment (jianjunz) [reply src/sdk/conference/client.js:444]: It's better to be `MediaStream` if it carries audio/video data, but it already defined.

File: src/sdk/conference/quicconnection.js (hunk 5 above)
Comment (jianjunz) [reply src/sdk/conference/quicconnection.js:137]: This is not a workaround. It's by design. Please see https://github.com/jianjunz/owt-server/blob/quic-protocol/doc/proposals/quic-transport-payload-format.md.

File: src/sdk/base/transport.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+     * one. It should always be undefined if transport type is webrtc since the
+     * webrtc agent of OWT server doesn't support multiple transceivers on a
+     * single PeerConnection.
+     */
+    this.id = id;
+  }
+}
+
+/**
+ * @class TransportSettings
```
Comment (taste1981) [inline src/sdk/base/transport.js:49]: Don't understand the definition of this class, besides this is not used anywhere.

Comment (jianjunz) [reply src/sdk/base/transport.js:49]: It's designed to be a property of publication or subscription.

File: src/sdk/conference/client.js (hunk 2 above)
Comment (jianjunz) [reply src/sdk/conference/client.js]: Removed.

File: src/sdk/conference/client.js (hunk 4 above)
Comment (jianjunz) [reply src/sdk/conference/client.js]: Fixed.

**Review by taste1981 — DISMISSED** [review]

**Review by taste1981 — APPROVED** [review]

**Pull request merged**